### PR TITLE
TCD/JDJ/RULE19-10

### PR DESCRIPTION
**Pull request by JacksonJ-KC**
@yunjoonjung-PNNL

### CODE REVIEW

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json [hunk 1]
```diff
@@ -0,0 +1,759 @@
...
+            "ruleset_reference": "G3.1.2.6",
+            "rule_description": "Air economizers shall be included in baseline HVAC Systems 3 through 8, and 11, 12, and 13 based on climate as specified in Table G3.1.2.6.\nExceptions:\n1. Systems that include gas-phase air cleaning to meet the requirements of Standard 62.1, Section 6.1.2. This exception shall be used only if the system in the  proposed design does not match the building design.\n2. Where the use of outdoor air for cooling will affect supermarket open refrigerated case-work systems. This exception shall only be used if the system in the  proposed design does not use an economizer. If the exception is used, an economizer shall not be included in the baseline building design.\n3. Systems that serve computer rooms complying with Section G3.1.2.6.1.",
+            "applicable_rmr": "Baseline RMR",
+            "rule_assertion": "=",
+            "comparison_value": "Expected Value",
+            "primary_rule": "Yes",
+            "schema_version": "0.0.23"
+        },
+        "rmr_transformations": {
+            "baseline": {
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:19]: Do we need ```proposed``` along with ```baseline```? Please also add ```proposed``` in the rest of the TCDs (e.g., (b), (c), (d), etc.).

**Review by yunjoonjung-PNNL — CHANGES_REQUESTED** [review]

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json (hunk 1 above)
Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:19]: Added

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json [hunk 2]
```diff
@@ -0,0 +1,1792 @@
...
+                                                        "id": "VAV Air Terminal 1",
+                                                        "is_supply_ducted": true,
+                                                        "type": "VARIABLE_AIR_VOLUME",
+                                                        "served_by_heating_ventilating_air_conditioning_system": "System 11"
+                                                    }
+                                                ],
+                                                "spaces": [
+                                                    {
+                                                        "id": "Space 1",
+                                                        "lighting_space_type": "OFFICE_ENCLOSED"
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:47]: Could you add the ```floor_area``` key (also in the following tests (e.g., (b), (c), (d), etc.)? This is needed for the ```is_economizer_modeled_in_proposed``` function.

Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:48]: Added

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json [hunk 3]
```diff
@@ -0,0 +1,1883 @@
...
+                                                        "served_by_heating_ventilating_air_conditioning_system": "System 5",
+                                                        "heating_source": "HOT_WATER",
+                                                        "heating_from_loop": "Boiler Loop 1"
+                                                    }
+                                                ]
+                                            }
+                                        ],
+                                        "heating_ventilating_air_conditioning_systems": [
+                                            {
+                                                "id": "System 5",
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:297]: ```id```s between baseline and proposed must be the same. Please also check the following rules (e.g., (c), (d), etc.).

Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:307]: This should be resolved now

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json [hunk 4]
```diff
@@ -0,0 +1,1883 @@
...
+                                                "spaces": [
+                                                    {
+                                                        "floor_area": 1000,
+                                                        "id": "Space 1",
+                                                        "lighting_space_type": "OFFICE_ENCLOSED"
+                                                    }
+                                                ]
+                                            },
+                                            {
+                                                "id": "Thermal Zone 2",
```
[the remark this answers is not in the document]
Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:284]: Resolved

**Comment by yunjoonjung-PNNL** [comment]
Looks great to me. Thank you!

**Review by yunjoonjung-PNNL — APPROVED** [review]

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json [hunk 5]
```diff
@@ -0,0 +1,1848 @@
...
+                                                            "id": "Return Fan 1"
+                                                        }
+                                                    ]
+                                                }
+                                            }
+                                        ],
+                                        "lighting_building_area_type": "RETAIL"
+                                    }
+                                ],
+                                "refrigerated_casess": [
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:1560]: Weili found a typo here. This is an error from the schema. So, could you replace ```refrigerated_casess``` with ```refrigerated_cases``` in both schema and 19_10.json instance?

Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_10.json:1560]: Yes, will do. I found the same typo as I was trying to validate this json file!

**Pull request merged**